File: EDUCATIONAL105/b.py
import math
import os
import random
import re
import sys


if __name__ == '__main__':
    filename = os.path.join(os.path.abspath(''), 'input.txt')
    read = open(filename)
    tests = int(read.readline())
    for t in range(tests):
        n, U, R, D, L = map(int, read.readline().split())
        for mask in range(16):
            rU, rR, rD, rL = U, R, D, L
            print(rU, rR, rD, rL, mask)
            if mask & 1:
                rU -= 1
                rL -= 1
            if mask & 2:
                rL -= 1
                rD -= 1
            if mask & 4:
                rD -= 1
                rR -= 1
            if mask & 8:
                rR -= 1
                rU -= 1
            if min(rU, rR, rD, rL) >= 0 and max(rU, rR, rD, rL) <= n - 2:
                print("YES")
                break
        else:
            print("NO")

File: EDUCATIONAL105/b.js
'use-strict'

const { PassThrough } = require("stream");

process.stdin.resume();
process.stdin.setEncoding('utf8');

let data = '';
let currentLine = 0;

process.stdin.on('data', chunk => {
data += chunk;
});

process.stdin.on('end', () =>{
    data = data.replace(/\s*$/, '')
    .split('\n')
    .map(str => str.replace(/\s*$/, ''));

    let testCases = parseInt(data.shift());

    while(testCases--) {

        const [N,U,R,D,L] = readLine().split(' ').map(Number);

        const res = b(N,U,R,D,L);

        console.log(res)
    }
});

function get_ints() { 
    return data.shift().split(' ').map(Number);
}
 
function readLine() { 
        return data[currentLine++];
}

function b(N,U,R,D,L){
    let [rU, rR, rD, rL] = [U,R,D,L]; 
    for(let mask = 0; mask <= 16; mask++){
        [rU, rR, rD, rL] = [U,R,D,L]; 
        if (mask & 1){
            rU-= 1;
            rL-= 1;
        }
        if (mask & 2){
            rL-= 1;
            rD-= 1;
        }
        if (mask & 4){
            rD-= 1;
            rR-= 1;
        }
        if (mask & 8){
            rR-= 1;
            rU-=1;
        }
        if (Math.min(rU,rR,rD,rL) >= 0 && Math.max(rU,rR,rD,rL) <= N - 2){
            return 'YES'
        }
    }
    return 'NO';
}
 
module.exports = b;